### PR TITLE
Remove code only used by ROCm < 5.0

### PR DESCRIPTION
**Pull request by Rombur**

### CODE REVIEW

File: core/src/HIP/Kokkos_HIP_Abort.hpp [hunk 1]
```diff
@@ -50,17 +50,7 @@
 
 #include <hip/hip_runtime.h>
 
-// FIXME_HIP ROCm 4.5 version header include would be <rocm/rocm_version.h>
-#if __has_include(<rocm_version.h>)
 #include <rocm_version.h>
```
Comment (masterleinad) [inline core/src/HIP/Kokkos_HIP_Abort.hpp:53]:
```suggestion
```
What's the reason for keeping this include file?

Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Abort.hpp:53]: I don't think you get the version if you don't include

Comment (masterleinad) [reply core/src/HIP/Kokkos_HIP_Abort.hpp:53]: Right, but we are not using the version anywhere anymore AFAICT. I don't care much either way but just want this to be a conscious choice.

Comment (Rombur) [reply core/src/HIP/Kokkos_HIP_Abort.hpp]: You are right, this is not needed anymore. I've removed it.

File: core/src/HIP/Kokkos_HIP_Abort.hpp [hunk 2]
```diff
@@ -50,18 +50,6 @@
 
 #include <hip/hip_runtime.h>
 
-// FIXME_HIP ROCm 4.5 version header include would be <rocm/rocm_version.h>
-#if __has_include(<rocm_version.h>)
-#include <rocm_version.h>
-#define KOKKOS_IMPL_ROCM_VERSION \
-  ROCM_VERSION_MAJOR * 10000 + ROCM_VERSION_MINOR * 100 + ROCM_VERSION_PATCH
```
Comment (dalg24) [inline core/src/HIP/Kokkos_HIP_Abort.hpp:57]: @Rombur you might want to include this in `HIP::print_configuration()` (see #5419 )

**Review by masterleinad — APPROVED** [review]

File: core/src/HIP/Kokkos_HIP_Abort.hpp (hunk 2 above)
Comment (masterleinad) [reply core/src/HIP/Kokkos_HIP_Abort.hpp:57]: If so, it would probably more sense to have it somewhere else, right?

Comment (dalg24) [reply core/src/HIP/Kokkos_HIP_Abort.hpp:57]: Yes

**Pull request merged**